src/scales/scale.time.js

```diff
@@ -61,253 +69,288 @@ module.exports = function(Chart) {
+			} else {
+				// No limit on the multiplier so it always fits
+				fits = true;
 			}
 
-			Chart.Scale.prototype.initialize.call(this);
-		},
-		getLabelDiff: function(datasetIndex, index) {
-			var me = this;
-			if (datasetIndex === null || index === null) {
-				return null;
+			if (fits) {
+				break;
 			}
+		}
 
-			if (me.labelDiffs === undefined) {
-				me.buildLabelDiffs();
+		return unit;
+	}
+
+	/**
+	 * Determines how we scale the unit
+	 * @param min {Number} the scale minimum
+	 * @param max {Number} the scale maximum
+	 * @param unit {String} the unit determined by the {@see determineUnit} method
+	 * @return {Number} the axis step size in milliseconds
+	 */
+	function determineStepSize(min, max, unit, maxTicks) {
+		// Using our unit, figoure out what we need to scale as
+		var unitDefinition = time[unit];
+		var unitSizeInMilliSeconds = unitDefinition.size;
+		var sizeInUnits = Math.ceil((max - min) / unitSizeInMilliSeconds);
+		var multiplier = 1;
+
+		if (unitDefinition.steps) {
+			// Have an array of steps
+			for (var i = 0; i < unitDefinition.steps.length && sizeInUnits > maxTicks; i++) {
+				multiplier = unitDefinition.steps[i];
+				sizeInUnits = Math.ceil((max - min) / (unitSizeInMilliSeconds * multiplier));
 			}
+		} else {
+			while (sizeInUnits > maxTicks) {
+				++multiplier;
+				sizeInUnits = Math.ceil((max - min) / (unitSizeInMilliSeconds * multiplier));
+			}
+		}
 
-			if (typeof me.labelDiffs[datasetIndex] !== 'undefined') {
-				return me.labelDiffs[datasetIndex][index];
+		return multiplier;
+	}
+
+	/**
+	 * Helper for generating axis labels.
+	 * @param generationOptions {ITimeGeneratorOptions} the options for generation
+	 * @param dataRange {IRange} the data range
+	 * @param niceRange {IRange} the pretty range to display
+	 * @return {Number[]} ticks
+	 */
+	function generateTicks(generationOptions, dataRange, niceRange) {
+		var ticks = [];
+		var stepSize = generationOptions.stepSize;
+		ticks.push(generationOptions.min !== undefined ? generationOptions.min : niceRange.min);
+		var cur = moment(niceRange.min);
+		while (cur.add(stepSize, generationOptions.unit).valueOf() < niceRange.max) {
+			ticks.push(cur.valueOf());
+		}
+		var realMax = generationOptions.max || niceRange.max;
+		var minSpacing = (dataRange.max - dataRange.min) / generationOptions.maxTicks;
+		var lastSpacing = realMax - ticks[ticks.length - 1];
+		if (lastSpacing >= minSpacing) {
+			ticks.push(realMax);
+		} else {
+			ticks[ticks.length - 1] = realMax;
+		}
+		return ticks;
+	}
+
+	/**
+	 * @function Chart.Ticks.generators.time
+	 * @param generationOptions {ITimeGeneratorOptions} the options for generation
+	 * @param dataRange {IRange} the data range
+	 * @return {Number[]} ticks
+	 */
+	Chart.Ticks.generators.time = function(generationOptions, dataRange) {
+		var niceMin;
+		var niceMax;
+		var isoWeekday = generationOptions.isoWeekday;
+		if (isoWeekday !== false) {
```

Comment:
I think this option only applies if `generationOptions.unit` is `'week'` otherwise it could be transforming data that is in seconds, or minutes.